### PR TITLE
Support loading/saving simdgroup matrix from threadgroup memory

### PR DESCRIPTION
**Pull request by maxwindiff**

### CODE REVIEW

**Comment by maleadt** [comment]
bors try

File: src/device/intrinsics/memory.jl [hunk 1]
```diff
@@ -78,6 +78,15 @@ Base.@propagate_inbounds Base.getindex(A::MtlLargerDeviceArray{T}, i1::Integer)
 Base.@propagate_inbounds Base.setindex!(A::MtlLargerDeviceArray{T}, x, i1::Integer) where {T} =
     arrayset(A, convert(T,x)::T, i1)
 
+Base.to_index(::MtlLargerDeviceArray{T}, i::Integer) where {T} = i
```
Comment (maleadt) [inline src/device/intrinsics/memory.jl:81]: Is this needed? I'm assuming you're copying the CUDA.jl perf optimization for Int32 indices here. That doesn't really belong in this PR, but other than that it should definitely have some comments.

Comment (maxwindiff) [reply src/device/intrinsics/memory.jl]: You're right it's not needed, I've removed it.
I copied it over from `src/device/array.jl` when I was trying to get `tg_a[x,y]` to work for `MtlLargerDeviceArray` (which is needed by the test).

**Comment by maleadt** [comment]
bors try
